substrate/frame/src/lib.rs

```diff
@@ -203,29 +203,40 @@ pub mod prelude {
 	/// Dispatch types from `frame-support`, other fundamental traits
 	#[doc(no_inline)]
 	pub use frame_support::dispatch::{GetDispatchInfo, PostDispatchInfo};
-	pub use frame_support::traits::{Contains, IsSubType, OnRuntimeUpgrade};
+	pub use frame_support::traits::{
+		Contains, EstimateNextSessionRotation, IsSubType, OnRuntimeUpgrade, OneSessionHandler,
```

Comment:
Which traits to include in the prelude is subjective I suppose, but adding these two here instead of just `use frame::traits::OneSessionHandler` might be adding too much if they're not used that often. In general the prelude should be sparing. I'll leave that decision to @kianenigma and @re-gius though since I haven't really been involved in this so far

Reply:
I agree with you that we should only include things that are used a lot in the prelude. Both `OneSessionHandler` and `OnRuntimeUpgrade` are used in various other pallets, hence I included them in the prelude. So my idea was to only import things from `frame::prelude` or `frame::deps` for dependencies. We can create modules to group similar ideas like `traits`, `arithmetic` or `cryptography` and re-export them in the prelude. This way all imports used in multiple pallets would be directly imported from the `prelude` and other imports from dependencies would be from `frame::deps`.
But given that I made this decision only based on the usage count of a particular type and there might be other considerations involved, happy to hear your thoughts on this @kianenigma and @re-gius.

Reply:
> In general the prelude should be sparing

My thinking so far has been to make preludes expansive: it will likely over-import, but hardly under-import.

To this particular case, we can defer the judgment to a bit of domain specific knowledge: 

`OnRuntimeUpgrade` is a no-brainer very commonly used trait, should be in prelude. 

`OneSessionHandler` is only for pallets that interact with `pallet-session`. All consensus/staking related pallets do so, but there are not so many of them. Let's leave this one out for now.

Reply:
Thanks for adding context to this. Based on what you said, I do agree with your approach in this case.
> I'd be happy to go either way in this one if someone has a strong opinion.

I just think that we should somehow concretely define what should and shouldn't be in the prelude. This would speed up the migration as there would be no ambiguity. For example, I was going about this based on usage count. If something is used in more than (let's say) 3 pallets, it goes in the prelude. But like you pointed out, in this case, it might make more sense to not have it in the prelude. Unfortunately, I don't have a suggestion on how we can define this other than the usage count itself.

Reply:
Let's keep them all in prelude, and stick to the `>=3` rule then :)